### PR TITLE
RavenDB-11909: Verify task are unchanged on DatabaseRecord update, verify CreateDatabaseOperation doesnt have tasks configurations

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 1]
```diff
@@ -1203,9 +1203,67 @@ private unsafe List<string> AddDatabase(TransactionOperationContext context, Bli
...
+                                        }
+                                    }
+                                    else if (databaseRecordAsJson.TryGet(task, out BlittableJsonReaderArray newDbRecordObject) && newDbRecordObject.Length > 0)
+                                    {
+                                        hasChanges = true;
+                                    }
+
+                                    if (hasChanges)
+                                        throw new RachisInvalidOperationException(
+                                            $"Cannot update {task} configuration with DatabaseRecord. Please use dedicated operation to update {task} configuration.");
```
Comment (grisha-kotler) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs]:
```suggestion
                                            $"Cannot update {task} configuration with DatabaseRecord. Please use a dedicated operation to update the {task} configuration.");
```

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs [hunk 2]
```diff
@@ -1203,9 +1203,67 @@ private unsafe List<string> AddDatabase(TransactionOperationContext context, Bli
...
+                            };
+
+                            if (dbDoc == null)
+                            {
+                                foreach (var task in tasksList)
+                                {
+                                    if (databaseRecordAsJson.TryGet(task, out BlittableJsonReaderArray dbRecordVal) && dbRecordVal.Length > 0)
+                                    {
+                                        throw new RachisInvalidOperationException(
+                                            $"Failed to create a new Database {addDatabaseCommand.Name}. Updating tasks configurations via DatabaseRecord is not supported, please use dedicated operation to update {task} configuration.");
```
Comment (grisha-kotler) [inline src/Raven.Server/ServerWide/ClusterStateMachine.cs]:
```suggestion
                                            $"Failed to create a new Database {addDatabaseCommand.Name}. Updating tasks configurations via DatabaseRecord is not supported, please use a dedicated operation to update the {task} configuration.");
```

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: done

File: src/Raven.Server/ServerWide/ClusterStateMachine.cs (hunk 2 above)
Comment (garayx) [reply src/Raven.Server/ServerWide/ClusterStateMachine.cs]: done

**Pull request merged**